### PR TITLE
Add event and visit tracking to the site using Ahoy

### PR DESCRIPTION
**Pull request by rememberlenny**
Breaking up previous #929 to identify issue.

## New Gems
Adding new gem: ```ahoy_matey```

## Database changes
Adding table for ```visits``` and ```ahoy_events```. Adding column on ```proposals```, ```reports``` and ```comments``` for visit tracking.

```visits``` and ```ahoy_events``` are used for tracking front-end and backend activity in relation to the logged on user. The events can be used to make details reports for error handling and observe user behavior.

```visit_id``` column is added to ```proposals```, ```reports``` and ```comments``` to associate the visit of a user, when new instances are created. For example, the visit behavior leading up to a proposal being created can be tracked and observed.

# How
*Magic*

---

But seriously, ```ahoy``` has great documentation that can be viewed here:

- [Ahoy repository](https://github.com/ankane/ahoy)

## Admin

Added two fields in ActiveAdmin for Tracking.

screen shot 2016-02-05 at 1 04 31 pm

### CODE REVIEW

**Comment by pkarman** [comment]
I will QA

File: db/migrate/20160208173646_add_the_visit_id_to_models.rb [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+    add_foreign_key :proposals, :visits
+    add_foreign_key :comments, :visits
+    add_foreign_key :reports, :visits
+  end
+
+  def down
+    remove_foreign_key :proposals, :visits
+    remove_foreign_key :comments, :visits
+    remove_foreign_key :reports, :visits
+    drop_column :proposals, :visit_id
```
Comment (pkarman) [inline db/migrate/20160208173646_add_the_visit_id_to_models.rb]: @rememberlenny this seems to need updating (again) -- should be `remove_column`

**Comment by pkarman** [comment]
QA:

* deployed to c2-dev
* previous issues described here: https://github.com/18F/C2/pull/936#issuecomment-182471500
* `/admin` events are still being logged. buttons have been fixed.
* data retention question not addressed.

@rememberlenny what are your thoughts on how long this data should be retained, and whether `/admin` events should be logged or not?

**Comment by rememberlenny** [comment]
@pkarman Thanks for taking a look at this!

For logs, I'd like to keep all the data if possible. If the table size does seem like an issue, we could always push the events onto another database/rails instance. For immediate, Im open to doing a 60 day retention policy. What do you think?

I think the admin events being logged is good. What do you think?

**Comment by pkarman** [comment]
I'm fine to keep everything (for now) and log admin events. Just wanted to make sure those were intentional.

**Pull request merged**